### ISSUE TITLE
Issues with finetune_trainer.py on multiple gpus

### ISSUE THREAD

**Issue by rabeehk** [huggingface/transformers#8718]
## Environment info     
- `transformers` version:3.5.1
- Platform: google cloud
- Python version: 3.7
- PyTorch version (GPU?): yes 1.6
- Tensorflow version (GPU?): - 
- Using GPU in script?: yes 
- Using distributed or parallel set-up in script?: - 

### Who can help
 Trainer: @sgugger
 Text Generation: @patrickvonplaten @TevenLeScao
 T5: @patrickvonplaten
 examples/seq2seq: @patil-suraj

## Information
Hi
I am trying to train finetune_trainer.py on multiple gpus, one issue I got is 

1) Runtime error: Input,ouptut,and indices must be on the current device, 
Looking into the codes, in training_args.py when you set device for the n_gpu > 0 on line 401, this should be changed from cuda:0 to cuda to me.

2) The accuracy on multiple gpus does not match on single gpu and this is much lower, any idea on this.

Thank you.

**Comment by sgugger** [comment]
Hi there. There is little anyone can do to help without knowing the actual command you are running.

**Comment by rabeehk** [comment]
Hi
I am running
https://github.com/huggingface/transformers/blob/master/examples/seq2seq/finetune_trainer.py
once
with multiple gpu machines once with 1 machine,
I have adapted it for my usecase, this is hard for me to provide the exact
command, as I need to share the whole codebase.
My question is though more general I observe performance differences if you
run this code on 1 dataset with multiple gpu/gpu, any thoughts on this?
thanks
Rabeeh
